src/optimality.js

```diff
@@ -32,7 +32,11 @@ export default function optimality(
     let notPset = setDifference(fullSet, Pset[Fset[j]]);
     if (notPset.length === 0) {
       Jset.push(Fset[j]);
-    } else if (W.selection(notPset, [Fset[j]]).max() <= 0) {
+    } else if (
+      Math.round(W.selection(notPset, [Fset[j]]).max() * 10000000000) /
+        10000000000 <=
+      0
```

Comment:
Can you write `1e10` instead?
Is there a reason for rounding to the 10th decimal? Isn't it the kind of cases where we should use `Number.EPSILON`?

Reply:
Here, I am looking to round to the n-th decimal where n is the number of significant digit from the input matrices. Is there a way to know the lowest number of digit of the entries of a given matrix (excluding zero which seem to be always a single digit) ?